portal-web/test/functional/com/liferay/portalweb/macros/Segmentation.macro

```diff
@@ -83,7 +99,11 @@ definition {
 	}
 
 	macro enableAssingRolesBySegmentsCheckbox {
-		Click(locator1 = "Segmentation#ENABLE_ASSIGN_ROLES_SEGMENTS_CHECKBOX");
+		if (IsNotChecked.isNotCheckedNotVisible(locator1 = "Segmentation#ENABLE_ASSIGN_ROLES_SEGMENTS_CHECKBOX")) {
+			Click.javaScriptClick(locator1 = "Segmentation#ENABLE_ASSIGN_ROLES_SEGMENTS_CHECKBOX");
+
+			AssertChecked.assertCheckedNotVisible(locator1 = "Segmentation#ENABLE_ASSIGN_ROLES_SEGMENTS_CHECKBOX");
```

Comment:
why you have this assert here?

Reply:
Sometimes it pass too fast the click and the click is not correctly done. I include the assert to confirm that the check is correctly checked.

Reply:
so... this looks like the case to you add one wait element appear... and after that add the click ...

Reply:
so it should be something like WaitForElementPresent(locator1 = "Segmentation#ENABLE_ASSIGN_ROLES_SEGMENTS_CHECKBOX"); before the click instead of the Assert?

Reply:
yep